js/admin/instructorNew.js

```diff
@@ -0,0 +1,221 @@
+
+// dialogs
+const dialogs = document.querySelectorAll('.mdc-dialog');
+dialogs.forEach(dialog => {
+ const modal = new mdc.dialog.MDCDialog(dialog);
+})
+
+// UI elements
+// get edit btns
+const cardList = document.getElementById('card-list');
+// get add btns
+const addBtn = document.getElementById('add-btn');
+
+// get the modal dialog box
+const modalDialog = document.getElementById('mdc-modal-dialog');
+
+// get all modal inputs
+const modalInputs = document.querySelectorAll('.modalInput');
+
+// function to init event listeners
+const initEvents = () => {
+  // to the card list, then delegate to the edit and delete btn
+  cardList.addEventListener("click", handleCardListClick);
+
+  // to the add btn
+  addBtn.addEventListener("click", handleAddInstructor);
+
+  // to the page
+  document.addEventListener("click", handleModalRemove);
+
+  // events for the inputs
+  modalInputs.forEach(modalInput => {
+    modalInput.addEventListener('focus', handleFocus);
+    modalInput.addEventListener('blur', handleBlur);
+  })
+}
+
+// when the cards are clicked 
+const handleCardListClick = (e) => {
+  // console.log(e.target.closest('button'));
+  const btn = e.target.closest('button');
+  
+  // if the element is null stop
+  if (btn === null ) return;
+
+  // check the btns title and carry out the operation required
+  if (btn.title === 'edit') {
+    const cardDetailSection = btn.parentElement.parentElement.previousElementSibling.children[1];
+
+    // get the instructor's data from the card
+    const instructor = extractCardDetail(cardDetailSection);
+
+    // console.log(instructor);
+    // open modal
+    handleModalOpening('Edit', instructor);
+  }else if (btn.title === 'delete' && confirm('You are about to remove the card')) {
+    // console.log('nay')
+    // get the card
+    const card = btn.parentElement.parentElement.parentElement.parentElement;
+
+    // remove it from
+    card.remove();
+  }
+}
+
+const extractCardDetail = (card) => {
+  const instructor = {};
+
+  // extracting the details
+  instructor.name = card.querySelector('.card__title').textContent;
+  instructor.role = card.querySelector('.card__subtitle').textContent;
+  instructor.bio = card.querySelector('.card__description').textContent;
+  instructor.githubID = card.querySelector('.githubID').textContent;
+  instructor.twitterID = card.querySelector('.twitterID').textContent;
+
+  return instructor;
+}
+// handle the add click
+const handleAddInstructor = (e) => {
+  handleModalOpening('Add');
+}
+
+const handleModalOpening = (action, instructor =  {
+  name: '',
+  role: '',
+  bio: '',
+  githubID: '',
+  twitterID: ''
+}) => {
+  // getting all the required details
+  // the header and btn texts
+  const modeTextSpaces = document.querySelectorAll('.dialog-mode');
+  const modeIcon = document.getElementById('update-icon');
+
+  modeIcon.className = 'fas fa-user-plus';
+  
+  // reveal the modal by removing the hide class
+  modalDialog.classList.remove('hide');
```

Comment:
here you could open the modal by referring to the appropriate item under `dialogInstances`, e.g.:
```js
dialogInstances.modal.open();
```
Assuming `data-modal-id` was set to `modal` for the modal.

Reply:
done